guild_scheduled_event.go

```diff
@@ -16,3 +25,301 @@ type GuildScheduledEvent struct {
 	Creator            *User       `json:"creator"`
 	UserCount          int         `json:"user_count"`
 }
+
+type GuildScheduledEventQueryBuilder interface {
+	WithContext(ctx context.Context) GuildScheduledEventQueryBuilder
+	WithFlags(flags ...Flag) GuildScheduledEventQueryBuilder
+
+	Gets(params *GetScheduledEvents) ([]*GuildScheduledEvent, error)
```

Comment:
Will this look like: `Client.GuildScheduledEvent(0).Gets(...)` ?

the current pattern is to put these one level up. Client.GetGuildScheduledEvents(), as this builder is specific to a single event.

Reply:
`Client.GuildScheduledEvent(0).Gets(...)` Using this pattern will allow global guild ID then the parameter of each method in will focus only for the event

Reply:
Looking at the api docs again. I noticed some things can be merged:

```go
Client.GuildScheduledEvent(0).Get(123) => Client.Guild(0).ScheduledEvent(123).Get()
```
```go
Client.GuildScheduledEvent(0).Create(...) => Client.Guild(0).CreateScheduledEvent(...)
```
```go
Client.GuildScheduledEvent(0).Update(123, ...) => Client.Guild(0).ScheduledEvent(123).Update(...)
```
```go
Client.GuildScheduledEvent(0).Delete(123) => Client.Guild(0).ScheduledEvent(123).Delete()
```
```go
Client.GuildScheduledEvent(0).GetMembers(123) => Client.Guild(0).ScheduledEvent(123).GetMembers()
```
```go
Client.GuildScheduledEvent(0).Gets(...) => Client.Guild(0).GetScheduledEvents()
```

I tend to group resources by the url. Discord docs are prone to changes, so I'm unaware if it's a guarentee to be the same in the future. This way it's more coherent with the current disgord api.

What do you think about ScheduledEvents being a part of GuildBuilder instead of it's own?

Reply:
Just ping me on Discord when you're ready. My email and github notifications are currently overflooded so I tend to miss updates at times

Reply:
okay, sorry not to update the code. get busy for this week. I'll try to update on the weekend.

Reply:
I've been updated this code, please recheck it, thank u @andersfylling